src/plugin/datasource.ts

```diff
@@ -15,6 +21,9 @@ export function WavefrontDatasource(instanceSettings, $q, backendSrv, templateSr
     this.backendSrv = new BackendSrvCancelledRetriesDecorator(backendSrv, $q);
     this.templateSrv = templateSrv;
     this.defaultRequestTimeoutSecs = 15;
+    const appId = instanceSettings.jsonData.cspOAuthClientId
+    const appSecret = instanceSettings.jsonData.cspOAuthClientSecret
+    const credentials = `Basic ${Buffer.from(`${appId}:${appSecret}`).toString('base64')}`;
 
```

Comment:
What are we trying to do here?
Buffer does not exist.

Reply:
Use btoa for base64 encoding.
Try this: `const credentials = `Basic ${btoa(`${appId}:${appSecret}`)};``